File: c#/WAX/StartTagWAX.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace WAXNamespace
{
    /**
     * This interface defines the methods that can be called
     * after the beginning of a start tag has been output, but it
     * (the start tag itself, not a terminating end tag) has been closed.
     * 
     * <p>Copyright 2008 R. Mark Volkmann</p>
     * <p>This file is part of WAX.</p>
     * <p>
     * WAX is free software.  You can redistribute it and/or modify it
     * under the terms of the GNU Lesser General Public License as published
     * by the Free Software Foundation, either version 3 of the License,
     * or (at your option) any later version.
     * </p>
     * <p>
     * WAX is distributed in the hope that it will be useful,
     * but WITHOUT ANY WARRANTY; without even the implied warranty
     * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     * See the GNU Lesser General Public License for more details.
     * </p>
     * <p>
     * You should have received a copy of the GNU Lesser General Public License
     * along with WAX.  If not, see http://www.gnu.org/licenses.
     * </p>
     * 
     * @author R. Mark Volkmann, Object Computing, Inc.
     */
    public interface StartTagWAX : ElementWAX {

        /**
         * @see WAX#attr(string, Object)
         */
        StartTagWAX attr(string name, Object value);

        /**
         * @see WAX#attr(string, string, Object)
         */
        StartTagWAX attr(string prefix, string name, Object value);

        /**
         * @see WAX#attr(bool, string, string, Object)
         */
        StartTagWAX attr(
            bool newLine, string prefix, string name, Object value);

        /**
         * @see WAX#Namespace(string)
         */
        StartTagWAX Namespace(string uri);

        /**
         * @see WAX#Namespace(string, string)
         */
        StartTagWAX Namespace(string prefix, string uri);

        /**
         * @see WAX#Namespace(string, string, string)
         */
        StartTagWAX Namespace(string prefix, string uri, string schemaPath);
    }
}

File: java/src/com/ociweb/xml/StartTagWAX.java
package com.ociweb.xml;

/**
 * This interface defines the methods that can be called
 * after the beginning of a start tag has been output, but it
 * (the start tag itself, not a terminating end tag) has been closed.
 * 
 * <p>
 *   Copyright (c) 2008, R. Mark Volkmann<br />
 *   All rights reserved.
 * </p>
 * <p>
 *   Redistribution and use in source and binary forms, with or without
 *   modification, are permitted provided that the following conditions are met:
 * </p>
 * <ul>
 *   <li>
 *     Redistributions of source code must retain the above copyright notice,
 *     this list of conditions and the following disclaimer.
 *   </li>
 *   <li>
 *     Redistributions in binary form must reproduce the above copyright notice,
 *     this list of conditions and the following disclaimer in the documentation
 *     and/or other materials provided with the distribution.
 *   </li>
 *   <li>
 *     Neither the name of Object Computing, Inc. nor the names of its
 *     contributors may be used to endorse or promote products derived
 *     from this software without specific prior written permission.
 *   </li>
 * </ul>
 * <p>
 *   This software is provided by the copyright holders and contributors "as is"
 *   and any express or implied warranties, including, but not limited to,
 *   the implied warranties of merchantability and fitness for a particular
 *   purpose are disclaimed. In no event shall the copyright owner or
 *   contributors be liable for any direct, indirect, incidental, special,
 *   exemplary, or consequential damages (including, but not limited to,
 *   procurement of substitute goods or services; loss of use, data, or profits;
 *   or business interruption) however caused and on any theory of liability,
 *   whether in contract, strict liability, or tort (including negligence
 *   or otherwise) arising in any way out of the use of this software,
 *   even if advised of the possibility of such damage.
 * </p>
 * @author R. Mark Volkmann, Object Computing, Inc.
 */
public interface StartTagWAX extends ElementWAX {

    /**
     * @see WAX#attr(String, Object)
     */
    StartTagWAX attr(String name, Object value);

    /**
     * @see WAX#attr(String, String, Object)
     */
    StartTagWAX attr(String prefix, String name, Object value);

    /**
     * @see WAX#attr(String, String, Object, boolean)
     */
    StartTagWAX attr(
        String prefix, String name, Object value, boolean newLine);

    /**
     * @see WAX#defaultNamespace(String)
     */
    StartTagWAX defaultNamespace(String uri);

    /**
     * @see WAX#defaultNamespace(String, String)
     */
    StartTagWAX defaultNamespace(String uri, String schemaPath);

    /**
     * @see WAX#defaultNS(String)
     */
    StartTagWAX defaultNS(String uri);

    /**
     * @see WAX#defaultNS(String, String)
     */
    StartTagWAX defaultNS(String uri, String schemaPath);

    /**
     * @see WAX#namespace(String, String)
     */
    StartTagWAX namespace(String prefix, String uri);

    /**
     * @see WAX#namespace(String, String, String)
     */
    StartTagWAX namespace(String prefix, String uri, String schemaPath);

    /**
     * @see WAX#ns(String, String)
     */
    StartTagWAX ns(String prefix, String uri);

    /**
     * @see WAX#ns(String, String, String)
     */
    StartTagWAX ns(String prefix, String uri, String schemaPath);

    /**
     * @see WAX#unescapedAttr(String, Object)
     */
    StartTagWAX unescapedAttr(String name, Object value);

    /**
     * @see WAX#unescapedAttr(String, String, Object)
     */
    StartTagWAX unescapedAttr(String prefix, String name, Object value);

    /**
     * @see WAX#unescapedAttr(String, String, Object, boolean)
     */
    StartTagWAX unescapedAttr(
        String prefix, String name, Object value, boolean newLine);
}
